templates/vue-demo-store/components/shared/SharedCountryStateInput.vue

```diff
@@ -0,0 +1,112 @@
+<script lang="ts" setup>
+const emit = defineEmits<{
+  (e: "update:countryId", value: string): void;
+  (e: "update:stateId", value: string): void;
+}>();
+
+const props = withDefaults(
+  defineProps<{
+    countryId: string;
+    stateId: string | null;
+    countryIdValidation?: object | null;
+    stateIdValidation?: object | null;
+  }>(),
+  {
+    countryId: "",
+    stateId: "",
+    countryIdValidation: null,
+    stateIdValidation: null,
+  }
+);
+
+const localCountry = ref(props.countryId);
+const localState = ref(props.stateId);
+
+const { getCountries, getStatesForCountry } = useCountries();
+
+const states = computed(() => {
+  return getStatesForCountry(localCountry.value);
+});
+
+const handleCountryUpdate = (e: Event) => {
+  const target = e.target as HTMLInputElement;
+  localCountry.value = target.value;
+  localState.value = "";
+  emit("update:stateId", "");
+  emit("update:countryId", target.value);
+};
+
+const handleStateUpdate = (e: Event) => {
+  const target = e.target as HTMLInputElement;
+  localState.value = target.value;
+  emit("update:stateId", target.value);
+};
```

Comment:
using useVModels would make it more clean without keeping additional local state https://vueuse.org/core/useVModels/#usevmodels
```suggestion
const { countryId, stateId } = useVModels(props, emit)

const { getCountries, getStatesForCountry } = useCountries();

const states = computed(() => {
  return getStatesForCountry(countryId.value);
});

watch(countryId, () => {
  stateId.value = '';
}) // maybe with 'immediate'
```

Reply:
Updated suggestion after the discussion:
```suggestion
const { countryId, stateId } = useVModels(props, emit)
const { getCountries, getStatesForCountry } = useCountries();
const states = computed(() => {
  return getStatesForCountry(countryId.value);
});

function onCountrySelectChanged() {
  stateId.value = '';
}
```